src/filters/matches.rs

```diff
@@ -0,0 +1,159 @@
+crate::include_proto!("quilkin.extensions.filters.matches.v1alpha1");
+
+use self::quilkin::extensions::filters::matches::v1alpha1::Matches as ProtoConfig;
+use crate::filters::{prelude::*, registry::FilterRegistry};
+
+pub const NAME: &str = "quilkin.extensions.filters.matches.v1alpha1.Matches";
+
+/// Creates a new factory for generating capture filters.
+pub fn factory() -> DynFilterFactory {
+    Box::from(MatchesFactory::new())
+}
+
+struct Matches {
```

Comment:
Let's add comments for this and the factory struct describing them since they're public in the rustdoc iirc

Reply:
`Match` could be a better name for the filter? think I remember a while ago @markmandel and I had discussed having a convention for filters and to use verbs like compress/encrypt where possible (not sure how well we've kept to it ^^)

Reply:
I was initially calling it `match`, the reason I initially went with matches is though is because `match` is a keyword in Rust, and I didn't want to to deal with any hassle that might come from that. Now that I've written it though, it should be easy enough to try and rename.

Reply:
I'll rename this in a follow-up PR with the addition of metrics since this PR is already big enough.